rust/lancedb/src/remote/db.rs

```diff
@@ -88,15 +107,24 @@ impl ConnectionInternal for RemoteDatabase {
             .await
             .unwrap()?;
 
-        let rsp = self
+        let req = self
             .client
             .post(&format!("/v1/table/{}/create/", options.name))
             .body(data_buffer)
             .header(CONTENT_TYPE, ARROW_STREAM_CONTENT_TYPE)
             // This is currently expected by LanceDb cloud but will be removed soon.
-            .header("x-request-id", "na")
-            .send()
-            .await?;
+            .header("x-request-id", "na");
+        let rsp = self.client.send(req).await?;
+
+        if rsp.status() == StatusCode::BAD_REQUEST {
+            let body = rsp.text().await?;
+            if body.contains("already exists") {
+                return Err(crate::Error::TableAlreadyExists { name: options.name });
```

Comment:
Pedantic nitpick unrelated to the PR: we could use HTTP 409 (the request is syntactically valid but can't be processed in the given server state) for "already exists" type errors instead of 400 (the request would never be valid)

Reply:
Yes I agree. I have quite a few nits for the API (why is everything a POST request?). It would be a lot easier to detect this error if we could detect from a status code.